packages/grid/_modules_/grid/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridSelector } from '../core/useGridSelector';
+import { useNativeEventListener } from '../../root/useNativeEventListener';
+import { GRID_SCROLL } from '../../../constants/eventsConstants';
+import { gridContainerSizesSelector } from '../../utils/sizesSelector';
+
+export const useGridInfiniteLoader = (
+  windowRef: React.MutableRefObject<HTMLDivElement | null>,
+  apiRef: GridApiRef,
+): void => {
+  const options = useGridSelector(apiRef, optionsSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
+  const isInScrollBottomArea = React.useRef<boolean>(false);
+
+  const handleGridScroll = React.useCallback(
+    (event) => {
+      const scrollPositionBottom =
+        event.target.scrollTop + containerSizes?.windowSizes.height + options.scrollBottomThreshold;
+
+      if (containerSizes && scrollPositionBottom >= containerSizes.dataContainerSizes.height) {
+        if (!isInScrollBottomArea.current && options.onScrollBottom) {
+          isInScrollBottomArea.current = true;
+          options.onScrollBottom();
```

Comment:
You should not call this one directly.
Our pattern is to publish an event then hook it with  `useGridApiEventHandler`
`  useGridApiEventHandler(apiRef, GRID_ROW_CLICK, options.onRowClick);`

So we can easily add more  listeners to this event using apiRef...

Reply:
But I need the custom handler to know when to fire it. It's not that it needs to be fired every time a scroll interaction happens. If you are talking about having a dedicated event called `onGridScrollBottom` and then emit and subscribe to it then that is different.

Reply:
yes for the second part.
BTW we don't need to put `grid` in our props name so we can do `onRowsScrollEnd` it's a bit nicer I think

Reply:
updated